lib/iris/coords.py

```diff
@@ -1240,13 +1240,20 @@ def xml_element(self, doc):
 
     def _xml_id(self):
         # Returns a consistent, unique string identifier for this coordinate.
-        unique_value = (self.standard_name, self.long_name, self.units,
-                        tuple(sorted(self.attributes.items())),
-                        self.coord_system)
+        unique_value = b''
```

Comment:
Not the prettiest of changes... but understandable. Did you look at something like a comprehension on the original unique_value which casts to utf-8?